modules/programs/obsidian.nix

```
@@ -0,0 +1,547 @@
+{
+  config,
+  pkgs,
+  lib,
+  ...
+}:
+let
+  inherit (lib)
+    mkOption
+    mkEnableOption
+    mkPackageOption
+    mkDefault
+    literalExpression
+    types
+    ;
+
+  cfg = config.programs.obsidian;
+
+  corePlugins = [
+    "audio-recorder"
+    "backlink"
+    "bookmarks"
+    "canvas"
+    "command-palette"
+    "daily-notes"
+    "editor-status"
+    "file-explorer"
+    "file-recovery"
+    "global-search"
+    "graph"
+    "markdown-importer"
+    "note-composer"
+    "outgoing-link"
+    "outline"
+    "page-preview"
+    "properties"
+    "publish"
+    "random-note"
+    "slash-command"
+    "slides"
+    "switcher"
+    "sync"
+    "tag-pane"
+    "templates"
+    "word-count"
+    "workspaces"
+    "zk-prefixer"
+  ];
+in
+{
+  options.programs.obsidian = {
+    enable = mkEnableOption "obsidian";
+    package = mkPackageOption pkgs "obsidian" { };
+
+    defaultSettings = {
+      app = mkOption {
+        description = "Settings to write to app.json.";
+        type = types.raw;
+        default = { };
+      };
+
+      appearance = mkOption {
```

```suggestion
      appSettings = mkOption {
        description = "Settings to write to app.json.";
        type = types.raw;
        default = { };
      };

      appearanceSettings = mkOption {
```
otherwise, maybe something like this ? So each file has its own `settings`

---

Continuing in previous conversation :)